ansible/idr-searchengine.yml

```diff
@@ -58,82 +60,148 @@
       ipam_config:
         - subnet=10.11.0.0/16
-      #networks_cli_compatible: yes
-      #state: present
 
-  #- name: Pull search engine docker image
-  #  docker_image:
-  #      name: "{{ search_engineicsearch_docker_image }}"
-  #      tag: latest
-  #      source: pull
+  - name: set vm.max_map_count to 262144 in sysctl
+    sysctl: name={{ item.varname }} value={{ item.varvalue }}
+    with_items:
+      - { varname: "vm.max_map_count", varvalue: "262144" }
 
-  - name: Run docker elasticsearch
+  - name: Run main docker elasticsearch main node
     become: yes
     docker_container:
       image: "{{ search_engineelasticsearch_docker_image }}"
-      name: searchengineelasticsearch
+      name: "searchengineelasticsearchnode1"
       cleanup: True
+      ulimits:
+        - 'memlock:-1:-1'
       env:
-         discovery.type: single-node
-         path.data: /var/lib/elasticsearch
-         path.logs: /var/log/elasticsearch
-         network.host:  0.0.0.0
-         cluster.name: docker-cluster
-         http.host: 0.0.0.0
-         #http.port: 9200
-         ES_JAVA_OPTS: "-Xmx4096m"
+        path.data: "/var/lib/elasticsearch"
+        path.logs: "/var/log/elasticsearch"
+        path.repo: "{{ elasticsearch_backup_folder }}"
+        node.name: searchengineelasticsearchnode1
+        bootstrap.memory_lock: "true"
+        network.host: 0.0.0.0
+        cluster.name: searchengine-cluster
+        cluster.initial_master_nodes: "{{ elasticsearchnodes | join(',') }}"
+        http.host: 0.0.0.0
+        #http.port: 9200
+        ES_JAVA_OPTS: "-Xms2g -Xmx2g"
+        ingest.geoip.downloader.enabled: "false"
       networks:
-      - name: searchengine-net
+      - name: "searchengine-net"
+      published_ports:
+        - "9201:9200"
+        - "9301:9300"
+      state: started
+      restart_policy: always
+      volumes:
+      - "{{ apps_folder }}/searchengine/elasticsearch/node1/data:/var/lib/elasticsearch"
+      - "{{ apps_folder }}/searchengine/elasticsearch/node1/logs:/var/log/elasticsearch"
+      - "{{ elasticsearch_backup_folder }}:{{ elasticsearch_backup_folder }}"
+
+  - name: Run docker elasticsearch for the remaining nodes
+    become: yes
+    docker_container:
+      image: "{{ search_engineelasticsearch_docker_image }}"
+      name: "searchengineelasticsearchnode{{ item }}"
+      ulimits:
+        - 'memlock:-1:-1'
+      cleanup: True
+      env:
+        path.data: "/var/lib/elasticsearch"
+        path.logs: "/var/log/elasticsearch"
+        path.repo: "{{ elasticsearch_backup_folder }}"
+        node.name: "searchengineelasticsearchnode{{ item }}"
+        bootstrap.memory_lock: "true"
+        network.host: 0.0.0.0
+        cluster.name: "searchengine-cluster"
+        discovery.seed_hosts: "searchengineelasticsearchnode1"
+        cluster.initial_master_nodes: "{{ elasticsearchnodes | join(',') }}"
+        http.host: 0.0.0.0
+        #http.port: 9200
+        ES_JAVA_OPTS: "-Xms1g -Xmx1g"
```

Comment:
Same as above, is the memory different between the main node and these ones expected?

Reply:
Again it is related to the machine's specifications. I think this should be fine, but if we got any issues in the future we may need to increase the RAM.